tests/conftest.py

```diff
@@ -125,14 +125,18 @@ def command_line(runtime: str, java_command_line: List[str], dotnet_command_line
             "--interpreted-frames-native-stack",
             str(CONTAINERS_DIRECTORY / "nodejs/fibonacci.js"),
         ],
+        "nodejs-attach": [
```

Comment:
I don't like adding it as a different runtime, it requires maintenance in many sites.

Instead, I suggest:
1. For the Dockerfile, I will get https://github.com/Granulate/gprofiler/pull/477 merged soon, this lets you easily add a separate image for `nodejs`. It could be implemented by passing a build arg that's used as an env var, and if the env var is present, the Dockerfile won't add the perf jit flags.
2. For `runtime_specific_args`, what makes more sense is to require the `profiler_type` fixture instead of using a separate runtime. `profiler_type` is not defined by default but I checked and all users of `runtime_specific_args` also define `profiler_type` in the test - it's a simple change to require `profiler_type` in `runtime_specific_args`, and to use its value to decide for `nodejs` which profiler mode is used (it's actually the exact value).
3. For `command_line`, I think it's used also in cases where `profiler_type` is not defined and that's just a "bug" in our test infra (there are tests which use it but don't define `profiler_type` and actually hardcode the value in the test, e.g this one: https://github.com/Granulate/gprofiler/blob/43e1e16042c758a987b1e4efb987829b86a93c74/tests/test_sanity.py#L87. I don't suggest we fix the test infra in this PR - as a fair workaround, we can parametrize the fixture in your test `test_nodejs_attach_maps` to the needed value. I'm opening a ticket for the long term fix in the test infra.

Reply:
Created https://github.com/Granulate/gprofiler/issues/484